### DISCUSSION TITLE
function ledcAttachChannel was not declared in this scope

### DISCUSSION THREAD

**Discussion by Hainer1** [espressif/arduino-esp32#9227] [Q&A]
Hi there,

I am using the ESP32-C6 development board, Arduino IDE 2.3 and ESP32 3.0.0.0 alpha 3. I need two different PWM frequencies, my plan was to use this function that worked in the past:
LEDCAttachChannel(SOUND_PIN_SPEAKER, SOUND_PWM_FREQ, SOUND_PWM_RESOLUTION, SOUND_PWM_CHANNEL):

But the problem is that ledcAttachChannel is now missing. 

Does anyone have any idea why this is not available and how I can work around this? 
Thank you very much.

**Comment by me-no-dev** [comment]
you no longer need to specify channel. this is all done for you in the background. https://docs.espressif.com/projects/arduino-esp32/en/latest/migration_guides/2.x_to_3.0.html#id2

**Comment by Hainer1** [comment]
No, the migration documentation points to the ledc page and ledcAttachChannel is clearly defined there.
However, I have tried it without defining the channels and it does not work.
My workaround now is to copy the old ledcAttachChannel function into the files esp32-hal-ledc.h & esp32-hal-ledc.c in
AppData\Local\Arduino15\packages\esp32\hardware\esp32\3.0.0-alpha3\cores\esp32

This works for me at the moment.

**Reply by P-R-O-C-H-Y** [reply]
@Hainer1 You have 2 options now in 3.0.0 (master branch) can use `ledcAttach(pin, freq, resolution)` or `ledcAttachChannel(pin, freq, resolution, channel)`. Both options are working well. There is also a pending PR with some fixes right now #9257.
